core/eolearn/tests/test_utils/test_common.py

```diff
@@ -50,3 +53,40 @@ def test_is_discrete_type(number_type, is_discrete):
         numpy_dtype = np.dtype(number_type)
 
     assert is_discrete_type(numpy_dtype) is is_discrete
+
+
+@pytest.mark.parametrize(
+    "map_function, data, spatial_axes, expected",
+    [
+        (partial(np.resize, new_shape=(2, 2)), np.zeros(shape=0), (1, 2), np.zeros((2, 2))),
+        (partial(np.resize, new_shape=(2, 2)), np.ones((2, 3, 4, 1)), (1, 2), np.ones((2, 2, 2, 1))),
+        (partial(np.resize, new_shape=(3, 2)), np.ones((2, 3, 4, 1)), (0, 1), np.ones((3, 2, 4, 1))),
+        (partial(np.resize, new_shape=(2, 3)), np.ones((2, 3, 4, 1)), (0, 2), np.ones((2, 3, 3, 1))),
+        (partial(np.resize, new_shape=(1, 2)), np.ones((2, 3, 4, 1)), (2, 3), np.ones((2, 3, 1, 2))),
+        (
+            partial(np.resize, new_shape=(2, 2)),
+            np.arange(2 * 3 * 4).reshape((2, 3, 4, 1)),
+            (1, 2),
+            np.array([[[[0], [1]], [[2], [3]]], [[[12], [13]], [[14], [15]]]]),
+        ),
+        (
+            partial(np.flip, axis=0),
+            np.arange(2 * 3 * 4).reshape((2, 3, 4, 1)),
+            (1, 2),
+            np.array(
+                [
+                    [[[8], [9], [10], [11]], [[4], [5], [6], [7]], [[0], [1], [2], [3]]],
+                    [[[20], [21], [22], [23]], [[16], [17], [18], [19]], [[12], [13], [14], [15]]],
+                ]
+            ),
+        ),
```

Comment:
i get lost in 'what is what' here frequently.
Find what is a common solution to this with `pytest`. Maybe `pytest.param` could help, maybe you have to extract the cases to a dataclass.

Reply:
Pytest.param applied hope it is less confusing.

Reply:
Extract the cases to a dataclass.